lib/webview-helpers.js

```diff
@@ -112,10 +162,34 @@ helpers.procFromWebview = async function procFromWebview (adb, webview) {
 // Get a list of available webviews by introspecting processes with adb, where
 // webviews are listed. It's possible to pass in a 'deviceSocket' arg, which
 // limits the webview possibilities to the one running on the Chromium devtools
-// socket we're interested in (see note on webviewsFromProcs)
-helpers.getWebviews = async function getWebviews (adb, deviceSocket) {
+// socket we're interested in (see note on webviewsFromProcs). We can also
+// direct this method to verify whether a particular webview process actually
+// has any pages (if a process exists but no pages are found, Chromedriver will
+// not actually be able to connect to it, so this serves as a guard for that
+// strange failure mode). The strategy for checking whether any pages are
+// active involves sending a request to the remote debug server on the device,
+// hence it is also possible to specify the port on the host machine which
+// should be used for this communication.
+helpers.getWebviews = async function getWebviews (adb, deviceSocket,
+  ensureWebviewsHavePages, webviewDevtoolsPort) {
+
   logger.debug('Getting a list of available webviews');
-  let webviews = await webviewsFromProcs(adb, deviceSocket);
+  let webviewProcs = await webviewsFromProcs(adb, deviceSocket);
+
+  if (ensureWebviewsHavePages) {
+    logger.info('Retrieved potential webviews; will filter out ones with no active pages');
+    webviewProcs = await asyncfilter(webviewProcs, async (wp) => {
+      return await webviewHasPages(adb, wp, webviewDevtoolsPort);
```

Comment:
return is redundant

Reply:
i don't think so, doesn't the filter method need to return true or false in order to do its job?

Reply:
I mean 

```
await asyncfilter(webviewProcs, async (wp) => await webviewHasPages(adb, wp, webviewDevtoolsPort))
```

should work just fine